### PR TITLE
mod: add Grey Order dtv round

### PR DESCRIPTION
**Pull request by CapitalBerserkTamra**
Talked with Droob about adding my own DTV round of the very old clan in cRPG1 The Grey Order.

The bots represent the playstyle used by GO the most, like Agility-focused shielders and Two-Handed users as well as very heavy Tincan builds - I didn't add ranged nor cavalry, I'd like to see how players fare against a late-map (after dwarves) melee only round which shifts from agility bots to strength bots.

2 lieutenants/bosses per wave named after very known Grey Order members.

Open to feedback and there might be errors in the code still.

Implementing a round like this might motivate other players/clans to do similar.

### CODE REVIEW

File: src/Module.Server/ModuleData/dtv/dtv_characters.xml [hunk 1]
```diff
@@ -3327,6 +3327,424 @@
...
+      <skill id="Crossbow" value="0" />
+      <skill id="Throwing" value="0" />
+    </skills>
+    <Equipments>
+      <EquipmentRoster>
+        <equipment slot="Head" id="Item.crpg_vaegir_helm_v2_h0" />
+        <equipment slot="Cape" id="Item.crpg_wolf_shoulder_v2_h0" />
+        <equipment slot="Body" id="Item.crpg_battania_woodland_chainmail_v3_h0" />
+        <equipment slot="Gloves" id="Item.crpg_plated_strip_gauntlets_v2_h0" />
+        <equipment slot="Leg" id="Item.crpg_plated_strip_boots_v2_h0"" />
```
Comment (Norr2) [inline src/Module.Server/ModuleData/dtv/dtv_characters.xml:3413]: There is an extra " on line:
3413

Comment (Norr2) [inline src/Module.Server/ModuleData/dtv/dtv_characters.xml:3413]: There is an extra " on line:
3413
3546
3585
3624
3664
3703
Almost all the lines with boots

Comment (Norr2) [reply src/Module.Server/ModuleData/dtv/dtv_characters.xml]: You missed lin 3546 :)

**Comment by namidaka** [comment]
> Talked with Droob about adding my own DTV round of the very old clan in cRPG1 The Grey Order.
> 
> The bots represent the playstyle used by GO the most, like Agility-focused shielders and Two-Handed users as well as very heavy Tincan builds - I didn't add ranged nor cavalry, I'd like to see how players fare against a late-map (after dwarves) melee only round which shifts from agility bots to strength bots.
> 
> 2 lieutenants/bosses per wave named after very known Grey Order members.
> 
> Open to feedback and there might be errors in the code still.
> 
> Implementing a round like this might motivate other players/clans to do similar.

This is indeed a good addition. I'll wait for droob to evaluate how hard the wave in order to estimate where to slot it in , as i have no specific knowledge of dtv waves.

**Review by Norr2 — APPROVED** [review]
Changes done

**Comment by CapitalBerserkTamra** [comment]
Tested it with Norr, it only crashed 3 times until it started working :)

File: src/Module.Server/ModuleData/dtv/dtv_class_divisions.xml [hunk 2]
```diff
@@ -1918,6 +1918,30 @@
       </Perk>
     </Perks>
   </MPClassDivision>
+
+    <MPClassDivision id="ccrpg_dtv_greyorder_balanced_infantry"
```
Comment (Norr2) [inline src/Module.Server/ModuleData/dtv/dtv_class_divisions.xml:1922]: Just noticed line 1922 here has "ccrpg..."

**Comment by Muparadzi** [comment]
@namidaka what are your thoughts on having bot names as direct references to players? On some other rounds we have slightly more obscure references such as Duke Rafael (rfl) and Droobert Humperdink. I'm thinking maybe we should follow suite with this round.

**Comment by CapitalBerserkTamra** [comment]
Just my thoughts:
In my opinion not allowing real player names would kind of remove the whole idea of having clan-specific rounds.

If people put in the work to have their clan represented in the DTV game mode while improving it; why not giving them that small goodie of having actual real player names represented.

I dunno if I could come up with meaningful synonyms.
Might just change the other bot names of rfl and co. in my opinion if that's an issue.

File: src/Module.Server/ModuleData/dtv/dtv_data.xml [hunk 3]
```diff
@@ -311,6 +311,33 @@
...
+    <Group classDivisionId="crpg_dtv_greyorder_heavy_two_handed" count="0.8" />
+	  <Group classDivisionId="crpg_dtv_greyorder_heavy_polearm" count="0.8" />
+	  <Group classDivisionId="crpg_dtv_greyorder_lieutenant_harpag" />
+	  <Group classDivisionId="crpg_dtv_greyorder_lieutenant_pawiu" />
+    </Wave>
+    <Wave>
+    <Group classDivisionId="crpg_dtv_greyorder_heavy_two_handed" count="2.0" />
+	  <Group classDivisionId="crpg_dtv_greyorder_heavy_polearm" count="2.0" />
+    <Group classDivisionId="crpg_dtv_greyorder_boss_tamra" />
+	  <Group classDivisionId="crpg_dtv_greyorder_boss_hetman" />
```
Comment (Muparadzi) [inline src/Module.Server/ModuleData/dtv/dtv_data.xml:337]: There are some inconsistencies with your tabs here

File: src/Module.Server/ModuleData/dtv/dtv_class_divisions.xml [hunk 4]
```diff
@@ -1870,6 +1870,270 @@
...
+  
+  <MPClassDivision id="crpg_dtv_greyorder_kuyak_two_handed"
+                   hero="crpg_dtv_greyorder_kuyak_two_handed"
+                   troop="crpg_dtv_greyorder_kuyak_two_handed"
+                   hero_idle_anim="act_vlandia_mp_voulgier_idle"
+                   troop_idle_anim="act_idle_spear_1_left_stance"
+                   multiplier="0"
+                   cost="0"
+                   icon="Special_Swordsman"
+                   melee_ai="70"
```
Comment (Muparadzi) [inline src/Module.Server/ModuleData/dtv/dtv_class_divisions.xml:1906]: I would bring the `melee_ai` down for all of your regular units. Tincans & dwarves have only 30 in comparison to your 70. Maybe make them 50 so they are still formidable in melee to account for the lack of ranged units.

**Review by Muparadzi — CHANGES_REQUESTED** [review]
Thanks for the submission, I think they are in a good place difficulty wise with some slight changes.

**Review by Muparadzi — DISMISSED** [review]

**Review by Muparadzi — APPROVED** [review]

**Pull request merged**